plugins/zenoh-plugin-trait/src/manager.rs

```diff
@@ -99,7 +106,7 @@ impl<StartArgs: PluginStartArgs, Instance: PluginInstance> DeclaredPlugin<StartA
 pub struct PluginsManager<StartArgs: PluginStartArgs, Instance: PluginInstance> {
     default_lib_prefix: String,
     loader: Option<LibLoader>,
-    plugins: Vec<PluginRecord<StartArgs, Instance>>,
+    plugins: HashMap<String, PluginRecord<StartArgs, Instance>>,
```

Comment:
`Vec` instead of `HashMap` was chosen because each plugin already knows it's unique name so it was excessive to store this name outside. As far as I see nothing changed, there is just `id` instead of `name`. Is there any particular reason to switch to `HashMap` ?
Storing plugins in `Vec` may become useful in hypothetical situation when plugins depends on each other and loading order is important. Though yes, at this moment choice between `Vec` and `HashMap` here is just a question of personal taste.